src/runtime.rs

```diff
@@ -0,0 +1,342 @@
+        "getRemainingTimeInMillis",
+        lambda_context.get_remaining_time_in_millis,
+    )?;
+    obj.prop("clientContext", lambda_context.client_context)?;
+    obj.prop("cognitoIdentityJson", lambda_context.cognito_identity_json)?;
+    Ok(obj)
+}
+fn get_module_and_handler_name() -> Result<(String, String), String> {
+    match get_handler_env() {
+        Some(handler) => {
+            let split: Vec<Option<String>> = handler
+                .split('.')
+                .map(|s| {
+                    if s.is_empty() {
+                        None
+                    } else {
+                        Some(String::from(s))
+                    }
+                })
+                .collect();
+            match split.len() {
+                2 => match (&split[0], &split[1]) {
+                    (Some(module_name), Some(handler_name)) => {
+                        Ok((module_name.into(), handler_name.into()))
+                    }
+                    _ => {
+                        Err(format!("Invalid handler name or LAMBDA_HANDLER env value: \"{}\": Should be in format {{filename}}.{{method_name}}", handler))
+                    }
+                },
+                _ => {
+                    Err(format!("Invalid handler name or LAMBDA_HANDLER env value: \"{}\": Should be in format {{filename}}.{{method_name}}", handler))
+                }
+            }
+        }
+        None => Err("env value LAMBDA_HANDLER is not set".to_string()),
+    }
+}
+
+fn get_task_root() -> String {
+    match get_env(LAMBDA_TASK_ROOT) {
+        Some(lambda_task_root) => lambda_task_root,
+        None => env::current_dir()
+            .unwrap()
+            .into_os_string()
+            .into_string()
+            .unwrap(),
+    }
+}
+
+fn get_hyper_client() -> HyperClient {
+    let pool_idle_timeout = get_pool_idle_timeout();
+    let https = hyper_rustls::HttpsConnectorBuilder::new()
+        .with_tls_config(TLS_CONFIG.clone())
+        .https_or_http()
+        .enable_http1()
+        .build();
+
+    let client = Client::builder(TokioExecutor::new())
+        .pool_idle_timeout(Duration::from_secs(pool_idle_timeout))
+        .pool_timer(TokioTimer::new())
+        .build(https);
+    client
+}
+
+fn get_pool_idle_timeout() -> u64 {
+    let pool_idle_timeout: u64 = env::var(environment::ENV_LLRT_NET_POOL_IDLE_TIMEOUT)
+        .map(|timeout| {
+            timeout
+                .parse()
+                .unwrap_or(DEFAULT_CONNECTION_POOL_IDLE_TIMEOUT_SECONDS)
+        })
+        .unwrap_or(DEFAULT_CONNECTION_POOL_IDLE_TIMEOUT_SECONDS);
+    if pool_idle_timeout > 300 {
+        warn!(
+            r#""{}" is exceeds 300s (5min), risking errors due to possible server connection closures."#,
+            environment::ENV_LLRT_NET_POOL_IDLE_TIMEOUT
+        )
+    }
+    pool_idle_timeout
+}
+
+fn get_value(headers: &HeaderMap, header: &HeaderName) -> String {
+    headers.get(header).unwrap().to_str().unwrap().to_string()
```

Comment:
Would panic if we have to header, return result instead. Also maybe rename to something more descriptive `get_header_value`?